### ISSUE TITLE
Training with `pre_trained` set to False, and bit_w, bit_a  are not None,  will cause the loss value become `nan`

### ISSUE THREAD

**Issue by phuocphn** [zhutmost/lsq-net#1]
First of all, I would like to thank zhutmost for such great implementation. Your source is very easy to modify and understand.
But I came to the problem that when I train a new quantized model from scratch (which pre_trained is set to `False`) and bit_w, bit_a is set to values in rang [2...16]. The loss value will become `nan` in every epoch. Could you give me some details about this problem and how to solve it ?

Thank you so much

**Comment by zhutmost** [comment]
Due to the coronavirus outbreak in China, I am still trapped at home and cannot go to the lab. So I cannot access my GPU sever to reproduce your problem.
To my knowledge, it is difficult to quantize a deep network from scratch. It will most likely fail to converge. Even if it can converge, it is easy to fall to a local optimal.
I recommend you to quantize your model from a pre-trained one. Actually, this strategy also appears in many existing works, including LSQ-Net.

**Comment by zhutmost** [comment]
You can try to change the hyper-parameters and the optimizer, which may help the model converge without pre-training. But I don't think it will obtain a better accuracy.

**Comment by zhutmost** [comment]
Since this issue is inactive for a long time, close it.
